matrix/max_area_of_island.py

```diff
@@ -0,0 +1,126 @@
+# Given an m x n binary matrix grid. An island is a group of 1's
+# (representing land) connected 4-directionally (horizontal or vertical.)
+# You may assume all four edges of the grid are surrounded by water.
+# The area of an island is the number of cells with a value 1 in the island.
+# Return the maximum area of an island in grid.
+# If there is no island, return 0.
+
+
+def is_safe(row: int, col: int, ROWS: int, COLS: int) -> bool:
+    """
+    Checking weather coordinate (row,col) is valid or not.
+
+    >>> is_safe(0, 0, 5, 5)
+    True
+
+    >>> is_safe(-1,-1, 5, 5)
+    False
+    """
+    return not (row < 0 or col < 0 or row >= ROWS or col >= COLS)
+
+
+def dfs(row: int, col: int, ROWS: int, COLS: int, seen: set) -> int:
+    """
+    Returns the current Area of island
+    >>> dfs(0,0, 8,8,set())
+    0
+    """
+    if is_safe(row, col, ROWS, COLS) and (row, col) not in seen and mat[row][col] == 1:
+        seen.add((row, col))
+        return (
+            1
+            + dfs(row + 1, col, ROWS, COLS, seen)
+            + dfs(row - 1, col, ROWS, COLS, seen)
+            + dfs(row, col + 1, ROWS, COLS, seen)
+            + dfs(row, col - 1, ROWS, COLS, seen)
+        )
+    else:
+        return 0
+
+
+def count_max_area(mat: list[list[int]]) -> int:
+    """
+    Finds the area of all islands and returns the maximum area.
+
+    >>> count_max_area([ \
+        [0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0], \
+        [0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0], \
+        [0, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0], \
+        [0, 1, 0, 0, 1, 1, 0, 0, 1, 0, 1, 0, 0], \
+        [0, 1, 0, 0, 1, 1, 0, 0, 1, 1, 1, 0, 0], \
+        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0], \
+        [0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0], \
+        [0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0]])
+    6
+    """
+
+    ROWS = len(mat)
+    COLS = len(mat[0])
+    seen: set = set()
```

Comment:
```suggestion
    seen = set()
```

Reply:
mypy throws following error if I remove `: set`.

```
matrix/max_area_of_island.py:59: error: Need type annotation for "seen" (hint: "seen: Set[<type>] = ...")
Found 3 errors in 1 file (checked 1043 source files)
```

Reply:
Yes, we generally don't use mypy here, just normal Python 3

Reply:
Okay, gotit
Removing :set from here